### PR TITLE
Integrate DS R1 GroupedTopk + Sigmoid Routing Into DS Routing

### PR DESCRIPTION
**Pull request by amirumoAMD**
## Motivation

Integrate new GroupedTopk kernel with support for score_mode sigmoid into routing + unified routing function, renamed function to reflect change. For changes done in https://github.com/ROCm/ATOM/pull/1044.

## Technical Details

grouped_topk kernel added to topk files, integrated into grouped topk.

## Test Plan

Pytest for kernel included in routing pytest file, lm_eval on atom deepseek r1 mxfp4

## Test Result

Kernel pytests passing, lm_eval 98% for atom integration for deepseek r1 mxfp4.

## Submission Checklist

- [x] Look over the contributing guidelines at https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

### CODE REVIEW

File: aiter/ops/triton/moe/moe_routing/routing.py [hunk 1]
```diff
@@ -285,76 +327,62 @@ def routing(logits, n_expts_act, sm_first=False):
...
+    # fused V4 path: fused routing math + sort (score_mode given)
+    # ------------------------------------------------------------------
+    n_shared = num_fused_shared_experts
+    n_expts_tot = n_routed + n_shared
+
+    if use_grouped_topk and num_expert_group != 1:
+        assert (
+            num_expert_group is not None and topk_group is not None
+        ), "use_grouped_topk requires num_expert_group and topk_group"
+        from aiter.ops.triton.moe.moe_routing.topk import grouped_topk
```
Comment (k50112113) [inline aiter/ops/triton/moe/moe_routing/routing.py]: can you move this to top of the file?

**Review by k50112113 — DISMISSED** [review]
Can you also help check if the _grouped_topk kernel you added here is functionally the same as the one on the HIP side? no need to add unit test

**Review by lburzawa — APPROVED** [review]

**Review by k50112113 — APPROVED** [review]
